cli/client.go

```diff
@@ -380,6 +387,86 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to parse compiled date: %w", err)
+		return nil
+	}
+
+	// compare to current version
+	appVersion := rconfig.Version
+
+	// skip check if stable & install is older than two week
+	if appVersion != "" && time.Since(dateCompiled) < time.Hour*24*7*2 {
+		return nil
+	}
+
+	// skip check if on dev/latest & install and less than 6wks old
+	if appVersion == "" && time.Since(dateCompiled) < time.Hour*24*7*6 {
+		return nil
+	}
+
+	v1, err := semver.NewVersion(appVersion)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to parse compiled version: %w", err)
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*80)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, releaseURL, nil)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to create request: %w", err)
+		return nil
+	}
+
+	client := http.DefaultClient
+	res, err := client.Do(req)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to get latest release information: %w", err)
+		return nil
+	}
+
+	resp := checkUpdateResponse{}
+	err = json.NewDecoder(res.Body).Decode(&resp)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to decode response: %w", err)
+		return nil
+	}
+	err = res.Body.Close()
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to close response body: %w", err)
+		return nil
+	}
+
+	v2, err := semver.NewVersion(resp.TagName)
+	if err != nil {
+		warningf(c.App.ErrWriter, "failed to parse latest version: %w", err)
+		return nil
+	}
+
+	if v1.LessThan(v2) {
+		warningf(c.App.ErrWriter, "version of Viam CLI is out of date, version %s available", resp.TagName)
```

Comment:
```suggestion
		warningf(c.App.ErrWriter, "Your CLI is out of date. Consider updating to version %s ", resp.TagName)
```

Brainstorming -- not perfect

Reply:
(iirc, strings displayed to users in our warnings can start with capital letters and be complete sentences.)